### DISCUSSION TITLE
put many files in the same bucket

### DISCUSSION THREAD

**Discussion by yamine99** [minio/minio#13722] [Q&A] [answered]
Hello everyone ! 
I want  put many files at same time in my server mini Io  ? do you have any idea ..how can i make it ! 
thanks in advance

**Comment by klauspost** [comment, marked as answer]
@yamine99 There is no S3 API call for that.

We do however have a "snowball" extension. It is described here: https://blog.min.io/minio-optimizes-small-objects/ - including how to upload a tar file that is unpacked into separate objects on upload.

This is only implemented in the Go SDK. See example here: https://github.com/minio/minio-go/blob/master/examples/minio/putobjectsnowball.go

**Comment by yamine99** [comment]
Thank you @klauspost  for your response !
